src/core/weechat.c

```diff
@@ -676,13 +676,13 @@ weechat_locale_check ()
 }
 
 /*
- * Callback for system signal SIGHUP: quits WeeChat.
+ * Callback for system signal SIGHUP: reloads config.
  */
 
 void
 weechat_sighup ()
 {
-    weechat_quit_signal = SIGHUP;
+    command_reload_files ();
```

Comment:
Here you should only set a variable, and not directly reload files, because this signal handler can be called at any time, so for example during a reload of config files.
To prevent any problem, this must be delayed and be done after in the main loop.

Reply:
How embarrassing! :)

Fixed.